test/manifest/linux-vmss.json

```diff
@@ -52,8 +52,7 @@
             }
         },
         "servicePrincipalProfile": {
-            "clientID": "{clientID}",
-            "secret": "{secret}"
+            "clientID": "msi"
```

Comment:
let's keep the sp test

Reply:
This modification is to solve the problem that blobfuse mount failed when authenticate by MSI. The root cause is MSI generated by aks-engine is not assigned to VMSS. 
BTW, considering sp is already tested by most jobs using the arm template [linux.json](https://github.com/kubernetes-sigs/blob-csi-driver/blob/master/test/manifest/linux.json), which set `useManagedIdentity: false` and 
```json
    "servicePrincipalProfile": {
        "clientID": "{clientID}", 
        "secret": "{secret}"
    }
```

And only job `pull-blob-csi-driver-e2e-vmss` uses template [linux-vmss.json](https://github.com/kubernetes-sigs/blob-csi-driver/blob/master/test/manifest/linux-vmss.json) which set `useManagedIdentity: true`. Maybe we should have set
```json
    "servicePrincipalProfile": {
        "clientID": "msi", 
    }
```
to test msi scenario?

Reply:
per the logs, `pull-blob-csi-driver-e2e` is using sp

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_blob-csi-driver/730/pull-blob-csi-driver-e2e/1557924587340042240/build-log.txt
```
[pod/csi-blob-controller-84cf49dc6d-zs47k/blob] I0812 03:13:38.637448       1 azure.go:104] read cloud config from file: /etc/kubernetes/azure.json successfully
[pod/csi-blob-controller-84cf49dc6d-zs47k/blob] I0812 03:13:38.638028       1 azure_auth.go:245] Using AzurePublicCloud environment
[pod/csi-blob-controller-84cf49dc6d-zs47k/blob] I0812 03:13:38.638061       1 azure_auth.go:130] azure: using client_id+client_secret to retrieve access token
```

`pull-blob-csi-driver-e2e-vmss` is using msi:
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_blob-csi-driver/730/pull-blob-csi-driver-e2e-vmss/1557924587390373888/build-log.txt

```
[pod/csi-blob-node-8trh8/blob] I0812 03:15:51.934107       1 azure_auth.go:245] Using AzurePublicCloud environment
[pod/csi-blob-node-8trh8/blob] I0812 03:15:51.934127       1 azure_auth.go:96] azure: using managed identity extension to retrieve access token
[pod/csi-blob-node-8trh8/blob] I0812 03:15:51.934131       1 azure_auth.go:102] azure: using User Assigned MSI ID to retrieve access token
[pod/csi-blob-node-8trh8/blob] I0812 03:15:51.934162       1 azure_auth.go:113] azure: User Assigned MSI ID is client ID. Resource ID parsing error: %+vparsing failed for 82e9bd91-c283-4b2d-8708-6d11b07d630d. Invalid resource Id format
```

Reply:
looks like you are right.